src/middlewared/middlewared/alembic/versions/12.0/2021-03-21_09-49_iscsi_target_extent_name_unique.py

```diff
@@ -0,0 +1,32 @@
+"""Adding unique constraint on iSCSITargetExtent(name)
+
+Revision ID: d015631e2929
+Revises: 90b815426c10
+Create Date: 2021-03-21 09:49:26.747269+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'd015631e2929'
+down_revision = '90b815426c10'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('services_iscsitargetextent', schema=None) as batch_op:
+        batch_op.create_unique_constraint('iscsi_target_extent_name_unique', ['iscsi_target_extent_name'])
```

Comment:
Thank you for your contribution! This migration will fail if existing table rows do not match the constraint. Although I understand that due to existing middleware validations it is highly unlikely, we must nevertheless at least rename such rows as failure here will prevent TrueNAS from upgrading and leave user with a broken system.

Reply:
You are totally right. But then, this should be a longer discussion. If for whatever reason two records exist with the same name, what is the expected operation now, and after the change? I saw uniqueness check in the code, that is why I added it. Is there a way in the upgrade procedure to do some pre-validation before the actual upgrade process? Perhaps, that would be the most reasonable way. So, for example if I rename iscsi targets, then it could affect/break services.

Reply:
Services probably never worked correctly with duplicate names, our goal here is only to avoid crashes on update.

Reply:
Ok, got it.